### PR TITLE
Consider init accessor in IsPropertyAccessorNameConflict

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs [hunk 1]
```diff
@@ -269,6 +270,10 @@ private bool IsSetPropertyAccessorNameConflict(SyntaxToken token)
                 => token.IsKind(SyntaxKind.SetKeyword)
                 && IsNameConflictWithProperty("set", token.Parent as AccessorDeclarationSyntax);
 
+            private bool IsInitPropertyAccessorNameConflict(SyntaxToken token)
+                => token.IsKind(SyntaxKind.InitKeyword)
+                && IsNameConflictWithProperty("init", token.Parent as AccessorDeclarationSyntax);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:275]: add test plz.

Comment (Youssef1313) [reply src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:275]: @CyrusNajmabadi I'll add one now.
After careful review, I believe I have a mistake here and the test should fail.
I'll confirm that first then fix the implementation here, which probably should be:

```suggestion
                && IsNameConflictWithProperty("set", token.Parent as AccessorDeclarationSyntax);
```

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs [hunk 2]
```diff
@@ -259,6 +259,7 @@ public override SyntaxToken VisitToken(SyntaxToken token)
 
             private bool IsPropertyAccessorNameConflict(SyntaxToken token)
                 => IsGetPropertyAccessorNameConflict(token)
+                || IsSetPropertyAccessorNameConflict(token)
```
Comment (davidwengier) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:262]: This is the same as line 263

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs [hunk 3]
```diff
@@ -259,6 +259,7 @@ public override SyntaxToken VisitToken(SyntaxToken token)
...
                 || IsSetPropertyAccessorNameConflict(token);
```
Comment (sharwell) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:263]:
```suggestion
                || IsInitPropertyAccessorNameConflict(token);
```

File: src/EditorFeatures/Test2/Rename/RenameEngineTests.CSharpConflicts.vb [hunk 4]
```diff
@@ -3604,6 +3604,28 @@ class C
...
+        Public Sub ConflictWhenRenamingPropertyInitterLikeMethod(host As RenameTestHost)
+            Using result = RenameEngineResult.Create(_outputHelper,
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true">
+                            <Document>
+class C
+{
+    public int MyProperty { get; {|conflict:init|}; }
+
     private void {|conflict:$$_set_MyProperty|}(int value) => throw null;
```
Comment (Youssef1313) [inline src/EditorFeatures/Test2/Rename/RenameEngineTests.CSharpConflicts.vb:3629]: Also adding @davidwengier as he worked on this a while ago.
The `init` accessor is compiled in the IL to `set_MethodName`.


```csharp
public class C {
    public int X {get; init;}
}
```

```il
    .property instance int32 X()
    {
        .get instance int32 C::get_X()
        .set instance void modreq(System.Runtime.CompilerServices.IsExternalInit) C::set_X(int32)
    }
```

@davidwengier Can you confirm if the added test is correct?

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs (hunk 2 above)
Comment (Youssef1313) [reply src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:262]: Fixed 😄 
Copy/paste is evil

File: src/EditorFeatures/Test2/Rename/RenameEngineTests.CSharpConflicts.vb (hunk 4 above)
Comment (davidwengier) [reply src/EditorFeatures/Test2/Rename/RenameEngineTests.CSharpConflicts.vb:3629]: Yes, the change suggested in your comment looks right to me.

**Review by davidwengier — APPROVED** [review]

**Comment by davidwengier** [comment]
Thanks as always @Youssef1313 👍

**Pull request merged**